### PR TITLE
Micro optimisations

### PR DESCRIPTION
**Pull request by billouboq**
Just cleaning and micro-optimizing some stuff, nothing extravagant.

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -311,7 +310,7 @@ exports.decodePayload = function (data, binaryType, callback) {
       }
 
       var ret = callback(packet, i + n, l);
-      if (false === ret) return;
+      if (ret) return;
```
Comment (darrachequesne) [inline lib/index.js]: The meaning has changed here, hasn't it? I think the `ret` response is used [here](https://github.com/socketio/engine.io-client/blob/1.8.2/lib/transports/polling.js#L131) to end the decoding earlier.

(it seems a test is missing :-) )

Comment (darrachequesne) [inline lib/index.js]: How about calling it `more`, instead of `ret`? (`if more === false...`)

Comment (billouboq) [inline lib/index.js]: will revert this change and call it more, will be more readable indeed.

**Comment by darrachequesne** [comment]
Thanks a lot! If you see anything else, do not hesitate.
